File: C++/Bubble_Sort.cpp
// Bubble Sort == Reapetedly swap two adjacent elements if they are in wrong order
// For n elements in array == (n-1) iterations to get sorted array
// If ith iteration == Check upto (n-i)
#include<iostream>
using namespace std;

int main()
{
    int n;
    cin >> n;

    int arr[n];


    for(int i=0;i<n;i++)
    {
        cin >> arr[i];
    }

    int counter = 1;
    while(counter < n-1)
    {
        for(int i=0;i<n-counter;i++)
        {
            if(arr[i]>arr[i+1])
            {
                // Swapping the numbers
                int temp = arr[i];
                arr[i] = arr[i+1];
                arr[i+1] = temp;
            }
        }
        counter++;
    }

    for(int i=0;i<n;i++)
    {
        cout << arr[i] << " ";
    }





    return 0;
}
 

File: Dart/bubble_sort.dart
class Bubble {
  List<int> sort(List<int> list) {
    final int length = list.length - 1;

    for (int i = 0; i < length; i++) {
      for (int j = 0; j < length - i; j++) {
        if (list[j] > list[j + 1]) {
          int aux = list[j];
          list[j] = list[j + 1];
          list[j + 1] = aux;
        }
      }
    }

    return list;
  }
}

void main() {
  //test algorithm with print, so sorry

  List<int> unsorted = [1, 3, 5, 2, 4, 6, 0];
  List<int> sorted = Bubble().sort(unsorted);

  print(sorted.toString());
}
